### PR TITLE
High contrast theme

### PR DESCRIPTION
**Pull request by edpratti**
This PR refactors the entire theme setup for PFP in order to add flexibility to both the dynamic theming and high contrast support.

We have to review edge cases to see how both of these features can collide.

# TODOs

- [x] Navigation items on the header becoming invisible due to hacky styling.

# Future Work

- [ ] Add E2E screenshot tests for high-contrast dark and light mode across browsers

## Summary by CodeRabbit

* **New Features**
  * Added text-style buttons with hover, pressed, focus, and disabled states.
  * Added configurable color scaling and improved high-contrast theme support.
  * Added sticker logo color variables.

* **Style**
  * Refreshed colors across cards, forms, navigation, search, events, quizzes, tooltips, embeds, and content views.
  * Updated typography, borders, backgrounds, and interaction states for improved visual consistency.
  * Improved markdown focus behavior and heading scroll positioning.

### CODE REVIEW

File: src/styles/markdown/base.scss [hunk 1]
```diff
@@ -17,26 +17,14 @@ body.dark,
...
-		// (tokens in src/views/base/navigation/header.module.scss)
-		//
-		// - this prevents elements targeted by the URL hash (e.g. "/some-page#Heading-ID")
-		//   from being obscured by the header
-		scroll-margin-top: calc(var(--page_header-height) + var(--site-spacing));
-	}
-
-	* {
-		scroll-margin-top: inherit;
-	}
```
Comment (fennifith) [inline src/styles/markdown/base.scss:39]: Why was this moved into the `h[...]` styles? There are non-heading elements (such as footnotes) that also need this margin.

**Review by fennifith — CHANGES_REQUESTED** [review]
https://github.com/playfulprogramming/playfulprogramming/pull/1632#discussion_r3818514555 seems like a valid concern - should this be addressed?

Also, can we resolve all the coderabbit comments after addressing them? Would make it a lot easier to filter out concerns that aren't valid.

**Review by fennifith — APPROVED** [review]
